content/blog/en/dojo-middleware.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Custom Dojo Middleware
+date: 2020-02-03
+author: Anthony Gubler & Rene Rubalcava
+---
+
+Dojo provides a [middleware system](https://dojo.io/learn/middleware/introduction) that you can use in developing widgets for your applications. There is a comprehensive list of [available middleware](https://dojo.io/learn/middleware/available-middleware) that you can use to manage local widget state, styling, or DOM related information.
+
+<!-- more -->
+
+Middleware is really interesting because they can be used to help you interact with the DOM or with the properties of your widget.
+
+Middleware is created with the same factory used to create a widget, except instead of returning a vnode, you return an object or function that defines the API of the middleware and can be used to do some extra work for your widget.
```

Comment:
```suggestion
Middleware gets created with the same factory used to create a widget, except instead of returning a `vnode`, an object or function gets returned defining the API of the middleware and can get used to do some extra work for your widget.
```